server/py/services/api/db/sqldb/db.py

```diff
@@ -5772,6 +5773,83 @@ def _delete_alert_notifications(
         if commit:
             session.commit()
 
+    def store_alert_activation(
+        self,
+        session,
+        alert_data: mlrun.common.schemas.AlertConfig,
+        event_data: mlrun.common.schemas.Event,
+        notifications_states: list[mlrun.common.schemas.NotificationState],
+    ):
+        extra_data = {
+            "notifications": [
+                notification.dict() for notification in notifications_states
+            ],
+            "criteria": alert_data.criteria.dict() if alert_data.criteria else None,
+        }
+
+        # For JOB entities, construct entity_id as "name.uid" format
+        if alert_data.entities.kind == mlrun.common.schemas.alert.EventEntityKind.JOB:
+            run_name = alert_data.entities.ids[0]
+            run_uid = event_data.value_dict.get("uid")
+            entity_id = f"{run_name}.{run_uid}"
+        else:
+            entity_id = alert_data.entities.ids[0]
+
+        alert_activation_record = AlertActivation(
+            id=alert_data.id,
```

Comment:
The `id` column is autogenerated by the DB when you insert a record, and the value is automatically increased by 1 on each insertion - it should not be manually set.
Furthermore, setting the activation id to the alert id is wrong, because they are 2 different things. A single alert can have multiple activations, and each activation will be a different record with a unique id.
```suggestion
```